tanimoto_gp/__init__.py

```diff
@@ -47,17 +47,27 @@ def predict_f(self, params: TanimotoGP_Params, smiles_test: list[str], full_cova
         else:
             K_test_test = jnp.ones((len(smiles_test)), dtype=float)
 
-        return kgp.noiseless_predict(
-            a=TRANSFORM(params.raw_amplitude),
-            s=TRANSFORM(params.raw_noise),
-            k_train_train=self._K_train_train,
-            k_test_train=K_test_train,
-            k_test_test=K_test_test,
-            y_train=self._y_train,
-            full_covar=full_covar,
-        )
+        if self._L_cached is not None:
+            return kgp._L_noiseless_predict(
+                a=TRANSFORM(params.raw_amplitude),
+                L=self._L_cached,
+                k_test_train=K_test_train,
+                k__test_test=K_test_test,
+                y_train=self._y_train,
+                full_covar=full_covar
+            )
```

Comment:
This is only correct if the `s/a` ratio from `params` matches the value used to generate the cache, is that correct?